### DISCUSSION TITLE
Is `if a or b` preferred over `if any([a, b])`?

### DISCUSSION THREAD

**Discussion by harupy** [astral-sh/ruff#7926] [Ideas]
```python
if a or b:  # b is not evaluated if a is truthy
   ...

# vs.

if any([a, b]):  # evaluates both a and b
   ...
```

`a` and `b` are arbitrary expressions. Does this deserve a new rule?

**Comment by Akshat111111** [comment]
A conditional expression can be used to verify if either an or b is true while optimizing the evaluation of b if an is already true. Here's an improved method:
```
result = a if a else b
if result:
    # code 
```

**Reply by hikaru-kajita** [reply]
Python Boolean operations [will perform short-circuiting](https://docs.python.org/3/library/stdtypes.html#boolean-operations-and-or-not) so probably `a or b` is better in terms of readability.
